File: Loenn/entities/theoRespawn.lua
local drawableSprite = require("structs.drawable_sprite")

local respawn = {}

respawn.name = "BrokemiaHelper/theoRespawn"
respawn.depth = 100
respawn.placements = {
    name = "respawn",
    data = {
        flag = ""
    }
}

-- Offset is from sprites.xml, not justifications
local offsetY = -10
local texture = "characters/theoCrystal/idle00"

function respawn.sprite(room, entity)
    local sprite = drawableSprite.fromTexture(texture, entity)

    sprite.y += offsetY
    sprite:setColor({160 / 255, 160 / 255, 160 / 255, 80 / 255})

    return sprite
end

return respawn

File: Ahorn/entities/theoRespawn.jl
module BrokemiaHelperTheoRespawn

using ..Ahorn, Maple

@mapdef Entity "BrokemiaHelper/theoRespawn" TheoRespawn(x::Integer, y::Integer, flag::String="")

const placements = Ahorn.PlacementDict(
    "Theo Respawn (Brokemia Helper)" => Ahorn.EntityPlacement(
        TheoRespawn
    )
)

sprite = "characters/theoCrystal/idle00"

function Ahorn.selection(entity::TheoRespawn)
    x, y = Ahorn.position(entity)

    return Ahorn.getSpriteRectangle(sprite, x, y - 10)
end

Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::TheoRespawn) = Ahorn.drawSprite(ctx, sprite, 0, -10, tint=((160, 160, 160, 70) ./ 255))

end